exp/controllers/awsmanagedmachinepool_controller.go

```diff
@@ -189,13 +195,23 @@ func (r *AWSManagedMachinePoolReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, r.reconcileDelete(ctx, machinePoolScope, managedControlPlaneScope)
 	}
 
-	return ctrl.Result{}, r.reconcileNormal(ctx, machinePoolScope, managedControlPlaneScope)
+	infraCluster, err := r.getInfraCluster(ctx, log, cluster, machinePoolScope.ManagedMachinePool)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("getting infra provider cluster or control plane object: %w", err)
+	}
+	if infraCluster == nil {
+		log.Info("AWSCluster or AWSManagedControlPlane is not ready yet")
+		return ctrl.Result{}, nil
+	}
+
+	return ctrl.Result{}, r.reconcileNormal(ctx, machinePoolScope, managedControlPlaneScope, infraCluster)
```

Comment:
I was wondering if we need to get the infra cluster again. Isn't the `infraCluster` already a property of `machinePoolScope`? Look at line 174.